warehouse/warehouse/src/warehouse_services.cpp

```diff
@@ -0,0 +1,170 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2012, Willow Garage, Inc.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of Willow Garage nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Dan Greenwald */
+
+#include <ros/ros.h>
+#include <moveit/warehouse/state_storage.h>
+#include <moveit_msgs/SaveRobotStateToWarehouse.h>
+#include <moveit_msgs/ListRobotStatesInWarehouse.h>
+#include <moveit_msgs/GetRobotStateFromWarehouse.h>
+#include <moveit_msgs/CheckIfRobotStateExistsInWarehouse.h>
+#include <moveit_msgs/DeleteRobotStateFromWarehouse.h>
+#include <moveit_msgs/RenameRobotStateInWarehouse.h>
+
+static const std::string ROBOT_DESCRIPTION="robot_description";
+
+bool storeState(moveit_msgs::SaveRobotStateToWarehouse::Request&  request,
+                 moveit_msgs::SaveRobotStateToWarehouse::Response& response,
+                 moveit_warehouse::RobotStateStorage* rs)
+{
+  const moveit_msgs::RobotState& state = request.state;
+  if ("" == request.name)
```

Comment:
Consider using request.name.empty() instead of constructing a new string, running a compare op and then destructing the string.